app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -2173,4 +2173,34 @@ public function addToCompany(Lead $lead, $company)
 
         return false;
     }
+
+    /**
+     * @param Lead $lead
+     * @param $score
+     *
+     * @return bool
+     */
+    public function scoreContactsCompany(Lead $lead, $score)
+    {
+        $success          = false;
+        $entities         = [];
+        $contactCompanies = $this->companyModel->getCompanyLeadRepository()->getCompaniesByLeadId($lead->getId());
+
+        if (!empty($contactCompanies)) {
+            foreach ($contactCompanies as $contactCompany) {
+                $company  = $this->companyModel->getEntity($contactCompany['company_id']);
+                $oldScore = $company->getScore();
+                $newScore = $score + $oldScore;
```

Comment:
This behavior is a little confusing based on the form's "Change company score" and the label "Company score." There's no indication based on label that the given score will be appended to the old. We probably want to use a different translation strings for this.